certora/specs/SocialRecoveryModule.spec

```
@@ -502,3 +503,113 @@ rule finalizeRecoveryAlwaysPossible(env e) {
 
     assert !isReverted, "legitimate recovery finalization reverted";
 }
+
+// This rule verifies that if recovery request data is changed, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - finalizeRecovery(...)
+// - cancelRecovery(...)
+// Each of these either updates or deletes the recovery request.
+rule recoveryRequestsChange(env e, calldataarg args, method f) {
+    uint i;
+    uint256 guardianApprovalCountBefore = currentContract.recoveryRequests[safeContract].guardiansApprovalCount;
+    uint256 newThresholdBefore = currentContract.recoveryRequests[safeContract].newThreshold;
+    uint64 executeAfterBefore = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersBefore = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    f(e, args);
+
+    uint256 guardianApprovalCountAfter = currentContract.recoveryRequests[safeContract].guardiansApprovalCount;
+    uint256 newThresholdAfter = currentContract.recoveryRequests[safeContract].newThreshold;
+    uint64 executeAfterAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+    address newOwnersAfter = currentContract.recoveryRequests[safeContract].newOwners[i];
+
+    assert (
+        guardianApprovalCountBefore != guardianApprovalCountAfter ||
+        newThresholdBefore != newThresholdAfter ||
+        executeAfterBefore != executeAfterAfter ||
+        newOwnersBefore != newOwnersAfter
+        ) =>
```

uber-formatting-nit:

```suggestion
    ) =>
```

---

Done 👍🏾